Hooks.ts

```diff
@@ -1,28 +1,58 @@
-import * as cloudflare from "@cloudflare/workers-types"
 import { http } from "cloudly-http"
 import { storage } from "cloudly-storage"
 import { Queue } from "./Queue"
+import { Types } from "./Types"
 
-export class Hooks {
-	private constructor(readonly queue: Queue) {}
-	async trigger(target: string, value: any): Promise<void> {
-		let request: http.Request
-		if (!http.Request.is(value)) {
-			request = http.Request.create({
-				header: {
-					contentType: "application/json",
-				},
-				method: "POST",
-				body: value,
-				url: target,
-			})
-		} else {
-			request = value
+export class Hooks<T extends string> {
+	#options?: Hooks.Options
+	private constructor(
+		readonly queue: Queue,
+		readonly registrations?: storage.KeyValueStore<{ url: string; header?: http.Request.Header }>
+	) {}
+	async register(listener: Hooks.Registration<T>): Promise<boolean> {
+		return (
+			this.registrations
+				?.set(`${listener.hook}|${new URL(listener.url).hostname}`, {
+					url: listener.url,
+					header: listener.header,
+				})
+				.then(
+					() => true,
+					() => false
+				) ?? false
+		)
+	}
+	configure(maxRetries: number, timeFactor: number) {
+		this.#options = { maxRetries, timeFactor }
+	}
```

Comment:
Why can't this be done through the constructor as well?

Reply:
I had an idea to provide the ability assign different retry amounts to different hooks, but left it like this. Should I remove this?

Reply:
Left the function, but made it extendable and chainable.